eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -3,93 +3,373 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
+# This script implements the resource management guidelines documented at https://github.com/Azure/azure-sdk-tools/blob/main/doc/engsys_resource_management.md
+
 #Requires -Version 6.0
 #Requires -PSEdition Core
+#Requires -Modules @{ModuleName='Az.Accounts'; ModuleVersion='1.6.4'}
+#Requires -Modules @{ModuleName='Az.Resources'; ModuleVersion='1.8.0'}
 
-[CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+[CmdletBinding(DefaultParameterSetName = 'Interactive', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
 param (
-    [Parameter(Mandatory = $true)]
-    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
-    [string] $ProvisionerApplicationId,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $ProvisionerApplicationId,
 
-    [Parameter(Mandatory = $true)]
-    [string] $ProvisionerApplicationSecret,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string] $ProvisionerApplicationSecret,
 
-    [Parameter(Mandatory = $true)]
-    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
-    [ValidateNotNullOrEmpty()]
-    [string] $TenantId,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $OpensourceApiApplicationId,
```

Comment:
Is there any better more compact way to pass these options instead of needing all 3?

Reply:
We could pass them in as an object? Unfortunately TenantId has to be passed in twice, because the tenant for the identity used for subscription cleanup may differ from the microsoft tenant used for the opensource API call (e.g. if we're cleaning up a dogfood cloud subscription).

Reply:
The values are stored as separate pipeline variables right now though, so if we used an object we would still have to construct one in the pipeline inline powershell.

Reply:
When we generate an AAD token from them is that a time limited token? If not then perhaps we generate the single token once and store it in KV.

Reply:
The token has a 24 hour lifetime.

Reply:
I'm thinking maybe we should introduce some sort of "AAD json config" model similar to the subscription configs that we can use. So instead of having separate kv secrets like `opensource-aad-app-id`, `opensource-aad-tenant`, we just have one `opensource-aad-app-config` and then a common helper for grabbing a token from a json config? Or something along those lines, perhaps a higher level abstraction for various credential values and token providers that reduces how many keyvault secrets we need.

Reply:
Sounds good that could also be a follow-up item though.